test/libyul/EVMCodeTransformTest.cpp

```diff
@@ -70,10 +72,12 @@ TestCase::TestResult EVMCodeTransformTest::run(std::ostream& _stream, std::strin
 
 	evmasm::Assembly assembly{CommonOptions::get().evmVersion(), false, std::nullopt, {}};
 	EthAssemblyAdapter adapter(assembly);
+	// use current version of dialect (default-constructed) intended, as code generation
+	// can be different from version to version
 	EVMObjectCompiler::compile(
 		*stack.parserResult(),
 		adapter,
-		CommonOptions::get().evmDialect(),
+		EVMDialect::strictAssemblyForEVMObjects(EVMVersion{}, std::nullopt),
```

Comment:
Even better, instead of repeating the dialect here, we should just use whatever is in the object:

```suggestion
		*stack.parserResult->dialect(),
```

Reply:
In fact, after recent refactors there's no need to pass the `Dialect` explicitly at all. `EVMObjectCompiler` should just take it from the `Object`. `EOFVersion` as well.

Reply:
good point!